src/__tests__/Button.js

```diff
@@ -37,4 +38,10 @@ describe('Button', () => {
     expect(json).toMatchSnapshot()
     expect(json).toHaveStyleRule('width', '100%')
   })
+
+  test('disabled prop sets', () => {
```

Comment:
Did `simulate('mouseEnter)` not work? I see you imported `enzyme` here but not using it. I'm assuming you wanted to mount the component and then test the styled on mouse enter?

Reply:
It not working.

Reply:
ohhh right! It wouldn't work because once you mount the component you can't use `toHaveStyleRule` anymore.

Reply:
Ah I think they actually added support for testing pseudo element's styles https://github.com/styled-components/jest-styled-components/pull/60/files#diff-99a96acd66a8852eaa09e5a093229cc9. Let's use that to test your hover styles

Reply:
Yep I tried it on your branch. Update to ` "jest-styled-components": "^4.7.1"` and then add tests for hover styles:
```
  test('disabled prop sets', () => {
    const json = renderer.create(<Button disabled theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.blue)
    expect(json).toHaveStyleRule('background-color', theme.colors.blue, {
      modifier: ':hover',
    })
  })

  test('without disabled prop sets', () => {
    const json = renderer.create(<Button theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.darkBlue, {
      modifier: ':hover',
    })
  })
```

Reply:
I updated my test. Thank you @msafari!!!!!!

Reply:
Yeah, generally we shouldn't ever need to use enzyme in this repo, since it's primarily CSS and styled-components - enzyme's really mostly useful for testing interactions with state